receiver/opencensusreceiver/octrace/opencensus.go

```diff
@@ -24,26 +24,53 @@ import (
 	commonpb "github.com/census-instrumentation/opencensus-proto/gen-go/agent/common/v1"
 	agenttracepb "github.com/census-instrumentation/opencensus-proto/gen-go/agent/trace/v1"
 	resourcepb "github.com/census-instrumentation/opencensus-proto/gen-go/resource/v1"
-	tracepb "github.com/census-instrumentation/opencensus-proto/gen-go/trace/v1"
 	"github.com/census-instrumentation/opencensus-service/data"
 	"github.com/census-instrumentation/opencensus-service/internal"
 	"github.com/census-instrumentation/opencensus-service/receiver"
 )
 
+const (
+	defaultNumWorkers = 4
+)
+
 // Receiver is the type used to handle spans from OpenCensus exporters.
 type Receiver struct {
-	spanSink receiver.TraceReceiverSink
+	spanSink    receiver.TraceReceiverSink
+	numWorkers  int
+	workers     []*receiverWorker
+	messageChan chan *traceDataWithCtx
+}
+
+type traceDataWithCtx struct {
+	data *data.TraceData
+	ctx  context.Context
 }
 
 // New creates a new opencensus.Receiver reference.
 func New(sr receiver.TraceReceiverSink, opts ...Option) (*Receiver, error) {
 	if sr == nil {
 		return nil, errors.New("needs a non-nil receiver.TraceReceiverSink")
 	}
-	oci := &Receiver{spanSink: sr}
+
+	messageChan := make(chan *traceDataWithCtx, 64)
+	oci := &Receiver{
```

Comment:
nit: not from your change but `oci` was when "interceptor" was used, but we changed it to "receiver", seems a good opportunity to change it.